src/main/java/lab7/reflections.md

```diff
@@ -0,0 +1 @@
+In my program, I used dependency injection by giving the searchRecipes method a DataService object instead of making one inside the method. This makes the code easier to test and change. In the tests, I used mocks by making fake data with lambda expressions, so I could test different situations. I used the DataService interface, which lets me use different versions of data services that all have the same method. This is called polymorphism — it means I can use different objects in the same way. This also helps decouple the code, so it doesn’t depend too much on one specific thing like a real database. It makes the program easier to test and update.
```

Comment:
You should mention that DataService is a functional interface, which is why a lambda expression can be used in place of the service